lib/dunmanifestin/palette.rb

```diff
@@ -27,11 +28,25 @@ def initialize raw_text, filename
 
   def name
     return file_basename if not titled?
-    lines.first[1..-1]
+    lines.first.sub(/\|(\w+).*$/, '\1')
   end
 
-  def sample genre, inflections = []
-    phrases.constrained_sample.reify(genre, inflections)
+  def sample genre, inflections = [], constraints = []
+    if inflections.any?
+      # don't use the churn if there are inflections, since
+      # they'll be cached...
+      phrases.sample.reify(genre, inflections)
+    else
+      if constraints.include? :recur
```

Comment:
Ok, so if the swatch is constrained to recur, we `#sample`, otherwise we `#generate`. That makes sense if `#recur` means "from the world," and not "from this phrase". Did I invent the second sense? I may have.